plugin/core/panels.py

```diff
@@ -48,8 +49,19 @@ class LspUpdatePanelCommand(sublime_plugin.TextCommand):
     """
 
     def run(self, edit, characters):
-        self.view.replace(edit, sublime.Region(0, self.view.size()), characters)
+        view = self.view
+        # get the text of the first line of every folded region
+        folded_regions = (view.substr(view.line(fr.a)) for fr in view.unfold(sublime.Region(0, view.size())))
+        view.replace(edit, sublime.Region(0, view.size()), characters)
+
+        for fr in folded_regions:
+            # get all file lines
+            for ln in view.find_by_selector("meta.diagnostic.preamble.lsp"):
+                if view.substr(ln) == fr:
+                    # fold the region spanning from the end of the line of the
+                    # filename to the to the end of the indented body
+                    view.fold(sublime.Region(ln.b, view.indented_region(ln.b + 2).b - 1))
```

Comment:
`indented_region` is an undocumented ST API function. Beware.

Reply:
as I have noticed by experimenting, indented_region returns the region of the bloc of code with the highest indentation level containing the point:
the point here is represented by the cursor:
![screenshot_1](https://user-images.githubusercontent.com/33235928/48441950-1b58a780-e78d-11e8-8467-c31c763b207d.png)
![screenshot_2](https://user-images.githubusercontent.com/33235928/48441951-1bf13e00-e78d-11e8-908f-2d395df4c4e1.png)
![screenshot_3](https://user-images.githubusercontent.com/33235928/48441952-1c89d480-e78d-11e8-8945-f9ffcb04a600.png)